Lib/fontTools/ttLib/tables/_n_a_m_e.py

```diff
@@ -1138,3 +1164,50 @@ def __repr__(self):
     150: 0,  # langAzerbaijanRoman → smRoman
     151: 0,  # langNynorsk → smRoman
 }
+
+
+class NameRecordVisitor(TTVisitor):
+    def __init__(self):
+        self.seen = set()
+
+
+@NameRecordVisitor.register_attrs(
+    (
+        (otTables.FeatureParamsSize, ("SubfamilyID", "SubfamilyNameID")),
+        (otTables.FeatureParamsStylisticSet, ("UINameID",)),
+        (
+            otTables.FeatureParamsCharacterVariants,
+            (
+                "FeatUILabelNameID",
+                "FeatUITooltipTextNameID",
+                "SampleTextNameID",
+                "FirstParamUILabelNameID",
+            ),
+        ),
+        (otTables.STAT, ("ElidedFallbackNameID",)),
+        (otTables.AxisRecord, ("AxisNameID",)),
+        (otTables.AxisValue, ("ValueNameID",)),
+        (otTables.FeatureName, ("FeatureNameID",)),
+        (otTables.Setting, ("SettingNameID",)),
+    )
+)
+def visit(visitor, obj, attr, value):
+    visitor.seen.add(value)
+
+
+@NameRecordVisitor.register(ttLib.getTableClass("fvar"))
+def visit(visitor, obj):
+    for inst in obj.instances:
+        if inst.postscriptNameID != 0xFFFF:
+            visitor.seen.add(inst.postscriptNameID)
+        visitor.seen.add(inst.subfamilyNameID)
+
+    for axis in obj.axes:
+        visitor.seen.add(axis.axisNameID)
+
+
+@NameRecordVisitor.register(ttLib.getTableClass("CPAL"))
+def visit(visitor, obj):
+    for nameID in obj.paletteLabels + obj.paletteEntryLabels:
+        if obj.version == 1:
```

Comment:
move this `if` out of the loop, it's constant. Maybe do the same as the code in the subsetter that you deleted:

```python
    if obj.version == 1:
        visitor.seen.update(obj.paletteLabels)
        visitor.seen.update(obj.paletteEntryLabels)
```